### PR TITLE
Editor: Remove post-editor feature flag

### PR DESCRIPTION
**Pull request by timmyc**
For #3676 - This branch removes all usage of the config flag `post-editor`.  Trying to keep PRs as small as possible for these flag removals, so I will follow up with other PRs for the remaining editor feature flags.

### To Test
Interact with various editor paths from calypso.  Open a new post to edit, click the edit button on a post in the posts list, and the ADD button in the sidebar and masterbar.

### CODE REVIEW

File: client/sections.js [hunk 1]
```diff
@@ -122,20 +122,18 @@ if ( config.isEnabled( 'reader' ) ) {
...
-	if ( config.isEnabled( 'post-editor/pages' ) ) {
-		editorPaths.push( '/page' );
-	}
-
-	sections.push( {
-		name: 'post-editor',
-		paths: editorPaths,
-		module: 'post-editor'
-	} );
+if ( config.isEnabled( 'post-editor/pages' ) ) {
```
Comment (mtias) [inline client/sections.js]: Do we need this one? I'd consider still keeping a feature check around the main section entry.

Comment (timmyc) [inline client/sections.js]: I was going to clean up that flag in a subsequent PR, but will remove this instance of it here.

Comment (timmyc) [inline client/sections.js]: simplified this logic in 1a193e5

**Comment by nylen** [comment]
Looks good & works well.  I can't think of a situation where we would want to disable the editor, so getting rid of these flags is a nice simplification.

**Pull request merged**